src/wrapped_radix_engine_toolkit/low_level/notary_signature_uniffi_fn.rs

```diff
@@ -24,18 +24,28 @@ pub fn notary_signature_get_signature(
 
 #[uniffi::export]
 pub fn android_notarize_hash_with_private_key_bytes(
-    private_key_bytes: Entropy32Bytes,
+    private_key_bytes: Exactly32Bytes,
     signed_intent_hash: &SignedIntentHash,
 ) -> Result<NotarySignature> {
-    let mut private_key_bytes = private_key_bytes;
-
     let ed25519_private_key =
-        Ed25519PrivateKey::from_bytes(private_key_bytes.to_bytes())?;
+        Ed25519PrivateKey::try_from(private_key_bytes.as_ref())?;
 
     let private_key = PrivateKey::from(ed25519_private_key);
     let signature = private_key.notarize_hash(signed_intent_hash);
-    private_key_bytes.zeroize();
-    // private_key.zeroize() // FIXME: Zeroize once RET has added Zeroize to PrivateKeys
+
+    Ok(signature)
+}
+
+#[uniffi::export]
+pub fn android_sign_hash_with_private_key_bytes(
+    private_key_bytes: Exactly32Bytes,
+    hash: &Hash,
+) -> Result<Signature> {
+    let ed25519_private_key =
+        Ed25519PrivateKey::try_from(private_key_bytes.as_ref())?;
+
+    let private_key = PrivateKey::from(ed25519_private_key);
```

Comment:
this step is not needed, you can use `sign` on `ed25519_private_key` (rename it -> `private_key`) directly. 

And you should return type to `Result<Ed25519Signature>` instead, and maybe change the whole func to:

```rust
{
    Ed25519PrivateKey::try_from(private_key_bytes.as_ref()).map(|pk| pk.sign(hash))
}
```